packages/manager/modules/pci/src/projects/project/storages/blocks/block/detach/detach.routing.js

```diff
@@ -1,13 +1,23 @@
 export default /* @ngInject */($stateProvider) => {
   $stateProvider
-    .state('pci.projects.project.storages.blocks.block.detach', {
-      url: '/detach',
-      component: 'pciProjectStorageBlocksBlockDetach',
+    .state('pci.projects.project.storages.blocks.detach', {
+      url: '/detach?storageId',
+      views: {
+        modal: {
+          component: 'pciProjectStorageBlocksBlockDetach',
+        },
+      },
       layout: 'modal',
       resolve: {
-        close: /* @ngInject */ ($state, projectId) => () => $state.go('pci.projects.project.storages.blocks', {
-          projectId,
-        }),
+        storageId: /* @ngInject */$transition$ => $transition$.params().storageId,
+        back: /* @ngInject */ ($rootScope, $state, projectId) => (reload = false) => {
+          if (reload) {
+            $rootScope.$emit('pci.projects.project.storages.blocks_refresh');
+          }
+          return $state.go('pci.projects.project.storages.blocks', {
+            projectId,
+          });
```

Comment:
Could it make sense to have a more global function/utility/resolve that takes the `states` as parameters ? This would avoid duplicating that logic which seems used quite a lot